contracts/Stage1/TokenManager.sol

```diff
@@ -1,58 +1,64 @@
+    /// @param _symbol The token symbol e.g. "WETH", "USDC", "USDT"
+    function get(string memory _symbol) external view returns(TokenData memory) {
+        for (uint256 i = 0; i < tokenSymbols.length; i++) if (cmpString(tokenSymbols[i], _symbol)) return tokenMetaData[_symbol];
+        revert("err-tok-not-found");
     }
 
-    // Get an array of all the accepted tokens
-    function getAcceptedTokens() external view returns (Token[] memory) {
-        return tokens;
+    function getAcceptedTokens() external view returns (string[] memory) {
+        return tokenSymbols;
     }
 
     // Add a token to the accepted list of tokens
-    /// @param _name 32-byte array value representation of the token symbol e.g. "WETH", "USDT"
     /// @param _addr the address of the token
-    /// @param _dec the decimals of the token
     /// @param _chainlinkAddr the address of the token / USD Chainlink data feed
     /// @param _chainlinkDec the number of decimals the Chainlink data feed uses
-    function addAcceptedToken(bytes32 _name, address _addr, uint8 _dec, address _chainlinkAddr, uint8 _chainlinkDec) public onlyOwner {
-        tokens.push(Token(_name, _addr, _dec, _chainlinkAddr, _chainlinkDec));
+    function addAcceptedToken(address _addr, address _chainlinkAddr, uint8 _chainlinkDec) public onlyOwner {
+        (string memory symbol, uint8 decimals) = getTokenMetaData(_addr);
+        tokenSymbols.push(symbol);
+        tokenMetaData[symbol] = TokenData(_addr, decimals, _chainlinkAddr, _chainlinkDec);
     }
 
     function deleteToken(uint256 index) private {
-        for (uint256 i = index; i < tokens.length - 1; i++) tokens[i] = tokens[i+1];
-        tokens.pop();
+        for (uint256 i = index; i < tokenSymbols.length - 1; i++) tokenSymbols[i] = tokenSymbols[i+1];
+        tokenSymbols.pop();
     }
 
     // Remove accepted token from accepted list of tokens
-    /// @param _name 32-byte array value representation of the token symbol e.g. "WETH", "USDT"
-    function removeAcceptedToken(bytes32 _name) public onlyOwner {
-        for (uint256 i = 0; i < tokens.length; i++) if (tokens[i].name == _name) deleteToken(i);
+    /// @param _symbol The token symbol e.g. "WETH", "USDT"
+    function removeAcceptedToken(string memory _symbol) public onlyOwner {
+        for (uint256 i = 0; i < tokenSymbols.length; i++) if (cmpString(tokenSymbols[i], _symbol)) deleteToken(i);
+        tokenMetaData[_symbol] = TokenData(address(0), 0, address(0), 0);
+    }
+
+    function cmpString(string memory a, string memory b) private pure returns (bool) {
+        return (keccak256(abi.encodePacked((a))) == keccak256(abi.encodePacked((b))));
+    }
+
+    function getTokenMetaData(address _addr) private view returns (string memory, uint8) {
+        string memory sym = ISeuro(_addr).symbol();
+        require(!cmpString(sym, ""), "err-empty-symbol");
+        uint8 dec = ISeuro(_addr).decimals();
```

Comment:
it seems a bit misleading to use the sEURO interface here, when this is just a route to the IERC20 interface. could we create another interface?

Reply:
i will import from OZ instead